src/NLog.Targets.Syslog/MessageCreation/SdElement.cs

```diff
@@ -15,30 +15,24 @@ internal class SdElement
         private static readonly byte[] RightBracketBytes = { 0x5D };
 
         private readonly SdId sdId;
-        private readonly IList<SdParam> sdParams;
+        private readonly List<SdParam> sdParams;
 
         public SdElement(SdElementConfig sdElementConfig, EnforcementConfig enforcementConfig)
         {
             sdId = new SdId(sdElementConfig.SdId, enforcementConfig);
             sdParams = sdElementConfig.SdParams.Select(sdParamConfig => new SdParam(sdParamConfig, enforcementConfig)).ToList();
         }
 
-        public static void Append(ByteArray message, IEnumerable<SdElement> sdElements, LogEventInfo logEvent)
+        public static void Append(ByteArray message, List<SdElement> sdElements, LogEventInfo logEvent)
         {
-            var elements = sdElements
-                .Select(x => new { SdId = x.sdId, RenderedSdId = x.sdId.Render(logEvent), SdParams = x.sdParams })
-                .ToList();
-
-            InternalLogDuplicatesPolicy.Apply(elements, x => x.RenderedSdId);
```

Comment:
Removing this users will not be able to see if there are duplicate SdIds which are causing them problems because forbidden by RFC 5424

Reply:
Was thinking that such an expensive check, that "only" wrote to the NLog InternalLogger was bad. But it seems you really like it. What if the check is only performed when the NLog InternalLogger is enabled?

Reply:
It would be great to be able to do the check only if the InternalLogger is enabled, but this would imply at least allocating a Dictionary.